### PR TITLE
refactors hooks and components

### PR DESCRIPTION
**Pull request by CalinR**

### CODE REVIEW

File: src/components/payment/withPaymentLogic.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        if (event?.data?.payload?.success) {
+          processOrder(iFrameWindow);
+        } else {
+          setIsProcessing(false);
+          event?.data?.payload && (
+            iFrameRef.current.style.height = `${event.data.payload.height}px`
+          );
+        }
+      }
+    });
```
Comment (travisblair) [inline src/components/payment/withPaymentLogic.js]: We should probably move this into a useEffect so we don't pile up listeners on re-render, and so we can cleanup on unmount.

```
useEffect(() => {
    let listener;
    window.addEventListener('message', listener = (event) => {
      if (event?.data?.responseType === 'PIGI_UPDATE_HEIGHT') {
        event?.data?.payload && (iFrameRef.current.style.height = `${event.data.payload.height}px`);
      }
      if (event?.data?.responseType === 'PIGI_ADD_PAYMENT') {
        if (event?.data?.payload?.success) {
          processOrder(iFrameWindow);
        } else {
          setIsProcessing(false);
          event?.data?.payload && (
            iFrameRef.current.style.height = `${event.data.payload.height}px`
          );
        }
      }
    });
    return () => window.removeEventListener('message', listener)
}, [])
```

File: src/helpers/address.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        });
+
+        return errors;
+      }
+
+      return null;
+    });
+};
+
+export const postAddress = async (addressEndpoint, countries, address, csrf) => {
```
Comment (travisblair) [inline src/helpers/address.js]: we can remove `async` as we don't appear to be using it
```suggestion
export const postAddress = (addressEndpoint, countries, address, csrf) => {
```

File: src/components/payment/withPaymentLogic.js (hunk 1 above)
Comment (CalinR) [reply src/components/payment/withPaymentLogic.js]: Good find. This has been fixed.

File: src/helpers/address.js (hunk 2 above)
Comment (CalinR) [reply src/helpers/address.js]: Yup, you are correct. This has now been removed.

File: src/helpers/address.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+};
+
+export const getCountryList = (countryInfo) => {
+  const unsortedCountries = Object.values(countryInfo);
+  // eslint-disable-next-line no-nested-ternary
+  const countries = unsortedCountries.sort((a, b) => ((a.name > b.name) ? 1 : ((b.name > a.name) ? -1 : 0)));
+  return countries;
+};
+
+export const verifyAddress = async (apiPath, countries, address, csrf) => {
```
Comment (travisblair) [inline src/helpers/address.js]:
```suggestion
export const verifyAddress = (apiPath, countries, address, csrf) => {
```

File: src/hooks/useCustomer.js [hunk 4]
```diff
@@ -5,12 +5,12 @@ import CheckoutContext from '../components/Context';
...
-    apiPath, csrf, setApplicationState, customer, setCustomer,
+    apiPath, csrf, setApplicationState, customer,
   } = useContext(CheckoutContext);
-  const [errors, setErrors] = useState();
-  const [isLoading, setLoading] = useState(false);
+  const [customerErrors, setCustomerErrors] = useState();
+  const [loadingCustomer, setLoadingCustomer] = useState(false);
 
-  const validateCustomer = useCallback(async () => fetch(`${apiPath}/validate_email_address?email_address=${customer.email_address}`, {
+  const validateCustomer = useCallback(async (data) => fetch(`${apiPath}/validate_email_address?email_address=${data.email_address}`, {
```
Comment (travisblair) [inline src/hooks/useCustomer.js]:
```suggestion
  const validateCustomer = useCallback((data) => fetch(`${apiPath}/validate_email_address?email_address=${data.email_address}`, {
```

File: src/hooks/useCustomer.js [hunk 5]
```diff
@@ -25,51 +25,45 @@ const useCustomer = () => {
         return response.errors;
       }
       return null;
-    }), [customer, csrf]);
+    }), [csrf]);
 
-  const submitCustomer = useCallback(async () => {
-    fetch(`${apiPath}/customer/guest`, {
+  const updateCustomer = useCallback(async (data) => {
```
Comment (travisblair) [inline src/hooks/useCustomer.js]:
```suggestion
  const updateCustomer = useCallback((data) => {
```

File: src/hooks/useDiscount.js [hunk 6]
```diff
@@ -9,16 +9,14 @@ const useDiscount = () => {
...
-  const [errors, setErrors] = useState();
-  const [isLoading, setLoading] = useState(false);
-  const [discount, setDiscount] = useState(applicationState.discounts[0]?.code ?? '');
-  const discountApplied = applicationState.discounts.length > 0;
-  const discountCode = applicationState.discounts[0]?.code;
+  const [discountErrors, setDiscountErrors] = useState();
+  const [loadingDiscount, setLoadingDiscount] = useState(false);
 
-  const validateDiscount = useCallback(async () => fetch(`${apiPath}/validate_discount_code?discount_code=${discount}`, {
+  const validateDiscount = useCallback(async (discount) => fetch(`${apiPath}/validate_discount_code?discount_code=${discount}`, {
```
Comment (travisblair) [inline src/hooks/useDiscount.js]:
```suggestion
  const validateDiscount = useCallback((discount) => fetch(`${apiPath}/validate_discount_code?discount_code=${discount}`, {
```

File: src/hooks/useDiscount.js [hunk 7]
```diff
@@ -33,9 +31,9 @@ const useDiscount = () => {
         return response.errors;
       }
       return null;
-    }), [discount, csrf]);
+    }), [csrf]);
 
-  const submitDiscount = useCallback(async () => {
+  const updateDiscount = useCallback(async (discount) => {
```
Comment (travisblair) [inline src/hooks/useDiscount.js]:
```suggestion
  const updateDiscount = useCallback((discount) => {
```

File: src/hooks/useLineItems.js [hunk 8]
```diff
@@ -9,13 +9,12 @@ const useLineItems = () => {
...
-    applicationState,
+    lineItems,
     setApplicationState,
   } = useContext(CheckoutContext);
-  const [isLoading, setLoading] = useState(false);
+  const [loadingLineItems, setLoadingLineItems] = useState(false);
 
-
-  const updateQuantity = useCallback(async (quantity, lineItemKey) => {
+  const updateLineItemQuantity = useCallback(async (quantity, lineItemKey) => {
```
Comment (travisblair) [inline src/hooks/useLineItems.js]: Not sure that this needs to be async / await as we are only fetching data and setting state in the resp.

**Review by travisblair — COMMENTED** [review]
Over all this is looks really good! Just a couple of nit pick suggestions

**Pull request merged**